### PR TITLE
Backup: support Azure and generic S3 backends

### PR DESCRIPTION
**Pull request by gsanchietti**
Card https://trello.com/c/vOW3IM9o/299-core-p2-cloud-backup-providers

API example:
```
api-cli run cluster/add-backup-repository --data '{"name": "az2", "provider": "azure", "password": "", "url": "azure:backup", "parameters": {"azure_account_name": "ns8test", "azure_account_key": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx}}'
```

Azure screenshots:
Screenshot from 2023-08-24 16-55-08

image

Generic S3 screenshots:
image

image

### CODE REVIEW

File: core/ui/public/i18n/en/translation.json [hunk 1]
```diff
@@ -390,7 +390,11 @@
     "repository_name": "Repository name",
     "repository_name_helper": "An easy to remember name for the repository",
     "url": "URL",
-    "url_placeholder": "bucket-name",
+    "azure_placeholder": "container-name",
+    "backblaze_placeholder": "bucket-name",
+    "aws_placeholder": "bucket-name",
+    "s3_placeholder": "minio.myserver.org/bucket-name",
+    "url_placeholder": "bucket, container name, or ",
```
Comment (andre8244) [inline core/ui/public/i18n/en/translation.json:397]:
```suggestion
    "url_placeholder": "bucket or container name",
```

File: core/ui/src/components/backup/AddRepositoryModal.vue [hunk 2]
```diff
@@ -582,16 +652,113 @@ export default {
...
+        if (isValidationOk) {
+          this.focusElement("name");
+          isValidationOk = false;
+        }
+      }
+      return isValidationOk;
     },
     validateAddAzureRepository() {
-      // clear errors
+            // clear errors
```
Comment (andre8244) [inline core/ui/src/components/backup/AddRepositoryModal.vue:708]:
```suggestion
      // clear errors
```

File: docs/ui/core.md [hunk 3]
```diff
@@ -151,6 +151,14 @@ podman exec -ti ns8-core yarn storybook
   - `yarn storybook`: start Storybook webapp with hot-reload
   - `yarn build`: compiles and minifies for production
 
+Then, copy `public/config/config.development.js.sample` to `public/config/config.development.js` and edit at least the `API_ENDPOINT` abd `WS_ENDPOINT` variables.
```
Comment (andre8244) [inline docs/ui/core.md:154]:
```suggestion
Then, copy `public/config/config.development.js.sample` to `public/config/config.development.js` and edit at least the `API_ENDPOINT` and `WS_ENDPOINT` variables.
```

**Review by andre8244 — COMMENTED** [review]
UI part LGTM, just a couple of minor fixes and typos

**Review by andre8244 — APPROVED** [review]

File: docs/ui/core.md [hunk 4]
```diff
@@ -151,6 +151,14 @@ podman exec -ti ns8-core yarn storybook
   - `yarn storybook`: start Storybook webapp with hot-reload
   - `yarn build`: compiles and minifies for production
 
+Then, copy `public/config/config.development.js.sample` to `public/config/config.development.js` and edit at least the `API_ENDPOINT` and `WS_ENDPOINT` variables.
+Finally, you need to disable the CORS check. Connect to the leader node and execute:
+```
+echo GIN_MODE=debug >> /etc/nethserver/api-server.env; systemctl restart api-server
+```
+
+You should now be able to access the development UI inside your browser at `http://localhost:8081`.
```
[the remark this answers is not in the document]
Comment (gsanchietti) [reply docs/ui/core.md:160]: The container listens on 8080 but VSCode maps it to 8081.

**Pull request merged**